### PR TITLE
[BUGFIX] Fix error when `RuntimeBatchRequest` is passed to `SimpleCheckpoint` with `RuntimeDataConnector`

### LINKED ISSUE #2647: batch_request passed in to SimpleCheckpoint errors
**Describe the bug**
When creating a `SimpleCheckpoint` and passing in a `batch_request: BatchRequest`, it seems to treat the `batch_request` as a dictionary instead of a `BatchRequest` and errors out `AttributeError: 'BatchRequest' object has no attribute 'items'` [here](https://github.com/great-expectations/great_expectations/blob/0.13.17/great_expectations/core/util.py#L71)

**To Reproduce**
```python


import great_expectations as ge
from great_expectations.core.batch import RuntimeBatchRequest
from great_expectations.cli.datasource import sanitize_yaml_and_save_datasource
from great_expectations.checkpoint import SimpleCheckpoint

import pandas as pd


context = ge.get_context()
df = pd.DataFrame({"col1": ["a", "a", "b", "c"], "col2": [1, 2, 3, 4]})
config = """
    name: my_pandas_datasource
    class_name: Datasource
    execution_engine:
        class_name: PandasExecutionEngine
    data_connectors:
        my_runtime_data_connector:
          class_name: RuntimeDataConnector
          batch_identifiers:
              - some_batch_identifier_so_this_can_work
"""



context.test_yaml_config(
    yaml_config=config
)
sanitize_yaml_and_save_datasource(context, config, overwrite_existing=True)


runtime_batch_request = RuntimeBatchRequest(
    datasource_name="my_pandas_datasource",
    data_connector_name="my_runtime_data_connector",
    data_asset_name="insert_your_data_asset_name_here",
    runtime_parameters={
      "batch_data": df
    },
    batch_identifiers={
        "some_batch_identifier_so_this_can_work": "blah",
    }
)
my_checkpoint = SimpleCheckpoint(
    name="my_checkpoint",
    data_context=context,
    batch_request = runtime_batch_request
)

>>

AttributeError                            Traceback (most recent call last)
<ipython-input-1-b72466658554> in <module>
     43     name="my_checkpoint",
     44     data_context=context,
---> 45     batch_request = runtime_batch_request
     46 )

~/.pyenv/versions/anaconda3-2020.02/envs/sandbox/lib/python3.7/site-packages/great_expectations/checkpoint/checkpoint.py in __init__(self, name, data_context, config_version, template_name, module_name, class_name, run_name_template, expectation_suite_name, batch_request, action_list, evaluation_parameters, runtime_configuration, validations, profilers, validation_operator_name, batches, site_names, slack_webhook, notify_on, notify_with, **kwargs)
    672             slack_webhook=slack_webhook,
    673             notify_on=notify_on,
--> 674             notify_with=notify_with,
    675         ).build()
    676 

~/.pyenv/versions/anaconda3-2020.02/envs/sandbox/lib/python3.7/site-packages/great_expectations/checkpoint/configurator.py in build(self)
    112         self._validate_slack_configuration()
    113 
--> 114         return self._build_checkpoint_config()
    115 
    116     def _build_checkpoint_config(self) -> CheckpointConfig:

~/.pyenv/versions/anaconda3-2020.02/envs/sandbox/lib/python3.7/site-packages/great_expectations/checkpoint/configurator.py in _build_checkpoint_config(self)
    134                         "config_version": self.other_kwargs.pop("config_version", 1.0)
    135                         or 1.0,
--> 136                         **self.other_kwargs,
    137                     }
    138                 )

~/.pyenv/versions/anaconda3-2020.02/envs/sandbox/lib/python3.7/site-packages/great_expectations/data_context/types/base.py in update(self, other_config, runtime_kwargs)
   1743                 updated_batch_request = nested_update(
   1744                     batch_request,
-> 1745                     other_batch_request,
   1746                 )
   1747                 self._batch_request = updated_batch_request

~/.pyenv/versions/anaconda3-2020.02/envs/sandbox/lib/python3.7/site-packages/great_expectations/core/util.py in nested_update(d, u, dedup)
     69 ):
     70     """update d with items from u, recursively and joining elements"""
---> 71     for k, v in u.items():
     72         if isinstance(v, Mapping):
     73             d[k] = nested_update(d.get(k, {}), v, dedup=dedup)

AttributeError: 'RuntimeBatchRequest' object has no attribute 'items'
```
**Expected behavior**
I expected the above to work since I'm passing in `batch_request: BatchRequest`

**Environment (please complete the following information):**
 - Operating System: MacOS
 - Great Expectations Version: 0.13.17

**Additional context**
None

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- https://github.com/great-expectations/great_expectations/issues/2647
- Fixes instances where a `RuntimeDataConnector` and `RuntimeBatchRequest` are passed to a `SimpleCheckpoint`
- Performs type checking before invoking `nested_update`, preventing TypeErrors that were impacting users.

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Comment by cdkini** [comment]
While we've determined the appropriate change to get things working, I want to do the following:
* Understand the circumstances around why this failed (`RuntimeDataConnector`, `SimpleCheckpoint`, etc ?)
* Write tests to cover these circumstances to prevent future issues

File: great_expectations/data_context/types/base.py [hunk 1]
```diff
@@ -1791,6 +1792,13 @@ def update(
 
                 other_batch_request = other_config.batch_request
 
+                # Necessary when using RuntimeDataConnector with SimpleCheckpoint
```
Comment (alexsherstinsky) [inline great_expectations/data_context/types/base.py]: @cdkini I agree with the logic of this change.  It seems to me that its place should be in the caller.  Based on my experience and understanding, all arguments to the constructors of "config" objects (such as `CheckpointConfig`) in `base.py` are already expected to be basic types (i.e., not typed objects, such as `BatchRequest` types).

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
Very nice direction.  Please see comment.  Thank you!

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Pull request merged**